src/components/Calculator.js

```diff
@@ -1,69 +1,68 @@
-      this.setState({ total, next, operation });
-    };
-    const { total, next, operation } = this.state;
-    const display = () => {
-      if (next) {
-        return next;
-      }
-      if (operation) {
-        return operation;
-      }
-      if (total) {
-        return total;
-      }
-      return 0;
-    };
-    return (
-      <div className="wrapper">
-        <input className="input-field" placeholder="0" type="text" maxLength="0" value={display()} onChange={() => ''} />
-        <div className="btn-wrapper">
-          <Btn value="AC" className="btn" click={handleClick} />
-          <Btn value="+" className="btn" click={handleClick} />
-          <Btn value="%" className="btn" click={handleClick} />
-          <Btn value="÷" className="btn light" click={handleClick} />
-        </div>
-        <div className="btn-wrapper">
-          <Btn value="7" className="btn" click={handleClick} />
-          <Btn value="8" className="btn" click={handleClick} />
-          <Btn value="9" className="btn" click={handleClick} />
-          <Btn value="x" className="btn light" click={handleClick} />
-        </div>
-        <div className="btn-wrapper">
-          <Btn value="4" className="btn" click={handleClick} />
-          <Btn value="5" className="btn" click={handleClick} />
-          <Btn value="6" className="btn" click={handleClick} />
-          <Btn value="-" className="btn light" click={handleClick} />
-        </div>
-        <div className="btn-wrapper">
-          <Btn value="1" className="btn" click={handleClick} />
-          <Btn value="2" className="btn" click={handleClick} />
-          <Btn value="3" className="btn" click={handleClick} />
-          <Btn value="+" className="btn light" click={handleClick} />
-        </div>
-        <div className="btn-wrapper">
-          <Btn value="0" className="btn last" click={handleClick} />
-          <Btn value="." className="btn dot" click={handleClick} />
-          <Btn value="=" className="btn light-end" click={handleClick} />
-        </div>
+  const handleClick = (buttonName) => {
+    const name = buttonName.target.innerText;
+    const results = calculate({ total, next, operation }, name);
+    setTotal(results.total);
+    setNext(results.next);
+    setOp(results.operation);
+  };
+
+  const display = () => {
+    if (next) {
+      return next;
+    }
+    if (operation) {
+      return operation;
+    }
+    if (total) {
+      return total;
+    }
+    return 0;
+  };
+  return (
+    <div className="wrapper">
+      <input className="input-field" placeholder="0" type="text" maxLength="0" value={display()} onChange={() => ''} />
+      <div className="btn-wrapper">
+        <Btn value="AC" className="btn" click={handleClick} />
+        <Btn value="+" className="btn" click={handleClick} />
+        <Btn value="%" className="btn" click={handleClick} />
+        <Btn value="÷" className="btn light" click={handleClick} />
```

Comment:
- [ ] It would be better to fix the value of the second button. It should be:

![calculator-plus-minus](https://user-images.githubusercontent.com/79876959/144552880-d14a8056-4ba0-4c4d-817f-a46a537ef96f.PNG)

It is necessary to fix it because you are calling for the wrong functionality (currently you are calling the sum) from `calculate.js`. ♻️